service/src/test/java/io/camunda/service/AdHocSubProcessActivityServicesTest.java

```diff
@@ -0,0 +1,294 @@
+import io.camunda.service.AdHocSubProcessActivityServices.AdHocSubProcessActivateActivitiesRequest.AdHocSubProcessActivateActivityReference;
+import io.camunda.service.security.SecurityContextProvider;
+import io.camunda.zeebe.broker.client.api.BrokerClient;
+import io.camunda.zeebe.broker.client.api.dto.BrokerResponse;
+import io.camunda.zeebe.gateway.impl.broker.request.BrokerActivateAdHocSubProcessActivityRequest;
+import io.camunda.zeebe.protocol.impl.record.value.adhocsubprocess.AdHocSubProcessInstructionRecord;
+import io.camunda.zeebe.protocol.record.value.AdHocSubProcessInstructionRecordValue.AdHocSubProcessActivateElementInstructionValue;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class AdHocSubProcessActivityServicesTest {
+
+  private static final String AD_HOC_SUB_PROCESS_INSTANCE_KEY = "123456";
+  private static final String ELEMENT_ID = "activity1";
+
+  private AdHocSubProcessActivityServices services;
+
+  @Mock
+  private BrokerClient brokerClient;
+
+  @Mock
+  private SecurityContextProvider securityContextProvider;
+
+  @Captor
+  private ArgumentCaptor<BrokerActivateAdHocSubProcessActivityRequest> requestCaptor;
+
+  @BeforeEach
+  public void before() {
+    final CamundaAuthentication authentication = CamundaAuthentication.none();
+    services =
+        new AdHocSubProcessActivityServices(brokerClient, securityContextProvider, authentication);
+  }
+
+  @Test
+  public void shouldActivateActivitiesWithEmptyVariables() {
+    // given
+    final AdHocSubProcessActivateActivityReference element =
+        new AdHocSubProcessActivateActivityReference(ELEMENT_ID, Map.of());
+    final AdHocSubProcessActivateActivitiesRequest request =
+        new AdHocSubProcessActivateActivitiesRequest(
+            AD_HOC_SUB_PROCESS_INSTANCE_KEY, List.of(element), true);
+
+    final AdHocSubProcessInstructionRecord expectedResponse =
+        new AdHocSubProcessInstructionRecord();
+    final BrokerResponse<AdHocSubProcessInstructionRecord> brokerResponse =
+        new BrokerResponse<>(expectedResponse);
+
+    when(brokerClient.sendRequest(requestCaptor.capture()))
+        .thenReturn(CompletableFuture.completedFuture(brokerResponse));
+
+    // when
+    final AdHocSubProcessInstructionRecord result = services.activateActivities(request).join();
+
+    // then
+    assertThat(result).isEqualTo(expectedResponse);
```

Comment:
❓ What's the point of this assertion? You are mocking the response and telling the broker client to return this on any request. I don't see the value of asserting the response is equal to what you just mocked.

Reply:
I simplified the test in 7a2774112ba5e3440ad7108823e8b067f3e1b448 and added a single test case checking the broker response is the one returned while omitting these assertions in the actual activation API tests.